internal/ci/runner.go

```diff
@@ -97,6 +97,39 @@ func (r *Runner) setup(run config.Run) error {
 			}
 		}
 	}
+	// running before-exec tasks
+	err := r.beforeExec(r.Project.BeforeExec)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r* Runner) execute(task string) (int, error) {
+	args := strings.Fields(task)
+	cmd := exec.Command(args[0], args[1:]...)
+	cmd.Stdout = r.Cli.Out()
+	cmd.Stderr = r.Cli.Out()
+	cmd.Dir = r.RunnerConfig.RootDir
+	cmd.Env = append(os.Environ())
+	err := cmd.Run()
+	if err != nil {
+		return 1, err
+	}
+	return 0, nil
+}
+
+func (r* Runner) beforeExec(tasks []string) (error) {
```

Comment:
Redundant parentheses.
```suggestion
func (r* Runner) beforeExec(tasks []string) error {
```